Editable Datetime for Dataframe

willhuang1997 opened the pull request:
<!--
Before contributing (PLEASE READ!)

⚠️ If your contribution is more than a few lines of code, then prior to starting to code on it please post in the issue saying you want to volunteer, then wait for a positive response. And if there is no issue for it yet, create it first.

This helps make sure:

  1. Two people aren't working on the same thing
  2. This is something Streamlit's maintainers believe should be implemented/fixed
  3. Any API, UI, or deeper architectural changes that need to be implemented have been fully thought through by Streamlit's maintainers
  4. Your time is well spent!

More information in our wiki: https://github.com/streamlit/streamlit/wiki/Contributing
-->

## 📚 Context

_Please describe the project or issue background here_

- What kind of change does this PR introduce?

  - [ ] Bugfix
  - [ ] Feature
  - [ ] Refactoring
  - [ ] Other, please describe:

## 🧠 Description of Changes

- _Add bullet points summarizing your changes here_

  - [ ] This is a breaking API change
  - [ ] This is a visible (user-facing) change

**Revised:**

_Insert screenshot of your updated UI/code here_

**Current:**

_Insert screenshot of existing UI/code here_

## 🧪 Testing Done

- [ ] Screenshots included
- [ ] Added/Updated unit tests
- [ ] Added/Updated e2e tests

## 🌐 References

_Does this depend on other work, documents, or tickets?_

- **Issue**: Closes #XXXX

---

**Contribution License Agreement**

By submitting this pull request you agree that all contributions to this project are made under the Apache 2.0 license.

sfc-gh-wihuang commented on frontend/src/components/widgets/DataFrame/columns/TimeColumn.ts:
```diff
@@ -0,0 +1,129 @@
+        )
+      }
+    },
+    getCellValue(cell: TimePickerCell): number | null {
```
I am wondering if this should return a number. When I was trying to return a number aka `new Date(cell.data.time).getTime()` though, the milliseconds would be off by the timezone... Might need to look into this more but it's possible the ISO string is a simpler solution?

sfc-gh-mnowotka committed:
Merge remote-tracking branch 'origin/hamburger-cleanup-take-2' into hamburger-cleanup-take-2

sfc-gh-mnowotka committed:
Remove changed snapshots.

sfc-gh-mnowotka committed:
Append special props only to the last dev menu item.

sfc-gh-mnowotka committed:
Add extra null check

sfc-gh-mnowotka committed:
Snapshots added

kasim-inan committed:
up version to 1.18.0 (#6090)

sfc-gh-mnowotka committed:
Remove default URL values where it doesn't make sense to keep them.

sfc-gh-kbregula committed:
fixup! fixup! Refactor st.code to be markdown independent

sfc-gh-kbregula committed:
fixup! fixup! fixup! Refactor st.code to be markdown independent

sfc-gh-mnowotka committed:
Added or modified tests are now migrated to the React Testing Libary as agreed.

sfc-gh-mnowotka committed:
Merge branch 'develop' into hamburger-cleanup-take-2

tconkling committed:
Add sfc-gh-mnowotka as a protobuf CODEOWNER (#6069)

kmcgrady committed:
Downgrade decamelize to avoid Safari break (#6094)

lukasmasuch committed:
Check availability of local storage (#6100)

lukasmasuch committed:
Improve readability of styled cells in dark mode (#6098)

* Use grey when background color is yellow

* Add test

* Remove snapshot

* Add updated snapshot

arnaudmiribel committed:
Update repo's README (#6016)

Co-authored-by: sfc-gh-skekre <snehan.kekre@snowflake.com>

lukasmasuch committed:
Improve type utils logging (#6113)

lukasmasuch committed:
Pin unixodbc version to fix CI workflows (#6123)

lukasmasuch committed:
Use initial widget value in case data editor was removed (#6111)

* New data editor component - part 1 (#6021)

* Add improvements

* Fix empty line

* Add test for getCategoricalOptions

* Add quiver tests

* Add improvements to dataframe handling

* Add tests for type utils

* Fix line changes

* Update type util tests

* Fix tests

* Renaming

* Add arrowUtils unit tests

* Add additional tests

* Add last unit test

* Add additional types

* Incorporated feedback

* Add additional description

* Add e2e tests for types

* Update e2e tests

* Lower wait time

* Add license header

* Add additional e2e tests

* Add empty index as fallback

* Remove default timeout for loading

* Add data editor e2e tests for input data

* Remove new line

* Remove decimal from unsupported types

* Add descriptions for pandas and numpy type

* Update comments

* Some refactoring

* Add comment

* Update comment

* Improve type utils and test

* Improvements to arrow utils

* Add interval types

* Extract pandas styler e2e tests

* Remove dataframe canvas rendering e2e tests

* Remove arrow dataframe canvas spec

* Update e2e spec numbers

* Improve type utils and e2e tests

* Add e2e tests for indices

* Add line again

* Fix typing issues

* Don't use themed snapshots

* Improve handling of empty state

* Add improvements for large data

* Fix e2e test

* Add styler e2e snapshots

* Add dataframe input screenshots

* Add data editor input data snapshots

* Hide toolbar in e2e tets

* Update e2e tests

* Various e2e fixes

* Remove unnecessary line

* Add snapshots

* Add missing snapshots

* Add parameter to keep trailing zeros

* Fix decimal types

* Remove data editor column type tests

* Remove invalid screenshot

* Add back correct screenshots

* Make data mocks consistent with last row None

* Update screenshots

* Remove columns parameter

* Add support for time formatting

* minor improvements

* Add PR feedback

* Add comments

* Remove snapshots

* Fix quiver tests

* Remove screenshots for regeneration

* Update snapshots

* Make proto changes back/forward compatible

* Move parameter up

* Add examples

* Added temporary workarounds

* Add new lines

* Regenerate snapshots

* Improve column config merge

* Docstring improvements for summary,num_rows,disabled,Returns

* Update data editor snapshots

* Fix issue with resizing not triggered on parameter change

* Split up column loader

* Fix tests

* Update snapshot

* Update test

* Add test for fromJson

* Add additional test cases

* Integrate PR review

---------

Co-authored-by: snehankekre <snehan.minerva@gmail.com>

lukasmasuch committed:
Add support for numeric inidices (#6122)

tconkling committed:
Fix member function caching (#6114)

We have a regression in caching - starting in 1.18, using `@st.cache_data` (or `@st.cache_resource`) on a class member function will fail with an error when the cached function was called:

```python
class TestClass:
    @cache_decorator
    def foo(_self):
        # We underscore-prefix `_self`, because our class is not hashable.
        return "it works!"

obj = TestClass()
result = obj.foo()  # Explode!
```

This is the result of a refactor that moved "cached function wrapper" logic into a class, `CachedFunc`, (from a closure). `functools.update_wrapper` (and `functools.wraps`) has unexpected behavior when applied to a class member function: the wrapped function no longer passes the "self" argument through to the underlying member function.

I don't know why this is! I couldn't find anything in the `functools` docs (or in a quick Google search) to explain the behavior. My guess is that using `update_wrapper` on a class instance ends up overwriting one or more instance properties that break its member function call semantics.

But the fix is simple: we hoist the `functools.wraps` call outside of `CachedFunc`, rather than using `functools` to modify the `CachedFunc` member function itself.

Fixes https://github.com/streamlit/streamlit/issues/6109

AnOctopus committed:
Fix detecting arrow df as widget when not editable (#6127)

The data editor introduced a new kind of state for an element, being a
widget only conditionally. Element replay code relied on the presence of
an id field in the protobuf for determining what is a widget, but arrow
dfs don't act like a widget, and don't call `register_widget`, unless
used with the data editor. In this situation, the id isn't actually set
on the protobuf, so we can use it being set for our widget detection.

Fixes #6103

tconkling committed:
cache_errors: don't refer to experimental_memo|singleton (#6128)

Our `UnserializableReturnValueError` cache error currently refers to our old cache decorator names (`@st.experimental_memo|singleton`), and our old experimental cache decorator docs. This PR updates the error text to refer to the new decorators and the new cache docs.

lukasmasuch committed:
Improve docstring (#6135)

kmcgrady committed:
Up version to 1.18.1 (#6101)

* up version to 1.18.0

* Downgrade decamelize to avoid Safari break (#6094)

* Check availability of local storage (#6100)

* Up version to 1.18.1

---------

Co-authored-by: kasim-inan <kasim.inan@snowflake.com>
Co-authored-by: Lukas Masuch <Lukas.Masuch@gmail.com>

sfc-gh-jgarcia committed:
Fix `st.metric` tooltip position regression (#6129)

* Fix regression on help tooltip position for st.metric

* Regenerate snapshots

* Add snapshots back

* Truncate text without the need for display flex on styled component

* Add new tests

* Add new screenshots

* Delete snapshots just in case

* updated snapshots

* clean up spec and test

---------

Co-authored-by: willhuang1997 <willhuang1997@gmail.com>

vdonato committed:
Support a new 'do-not-merge' label (#6139)

sfc-gh-kbregula committed:
Fix tests

snehankekre committed:
Embed `st.experimental_data_editor` apps in docstring (#6144)

* Embed `st.experimental_data_editor` apps in docstring

* Enclose inline code in reStructuredText within two backticks

lukasmasuch committed:
Improve expander overflow (#6148)

sfc-gh-tszerszen committed:
Introduce make frontend-fast command (#6172)

sfc-gh-mnowotka committed:
Merge pull request #6080 from streamlit/hamburger-cleanup-take-2

Hamburger menu cleanup

sfc-gh-tszerszen committed:
Fix ?embed and ?embedded params

AnOctopus committed:
Remove frontend dependencies codeowners (#6181)

We added the frontend dependencies file to ensure licenses for new
dependencies would be properly checked, but now we have a check for it
in CI.

kmcgrady committed:
Up version to 1.19.0 (#6179)

Co-authored-by: Ken McGrady <ken.mcgrady@snowflake.com>

sfc-gh-tszerszen committed:
Add testing-library/jest-dom to our dependencies (#6175)

AnOctopus committed:
Allow overriding timeout arg for running a script (#6173)

3s is plenty for small scripts that test Streamlit feature
implementations, but not for testing apps.

sfc-gh-tszerszen committed:
Allow tests bo to filtered by file (#6169)

sfc-gh-mnowotka committed:
Merge pull request #6011 from streamlit/fix-embed-and-embedded-params

Fix ?embed and ?embedded params

sfc-gh-jgarcia committed:
Printing adjustments (#6180)

lukasmasuch committed:
Fix dtype check (#6188)

* Use kind for checking type

* Added another complex type in series

sfc-gh-tszerszen committed:
Disallow setting embed query param using set_query_param method (#6019)

sfc-gh-mnowotka committed:
Merge pull request #5969 from streamlit/sfc-gh-kbregula-http-support

Add SSL support

snehankekre committed:
Clearly state we don't guarantee mapbox token will work indefinitely (#6143)

kajarenc committed:
Fix pandas version <2 (#6202)

sfc-gh-jgarcia committed:
Fix on st.metric tooltip positioning (#6168)

* Fix on st.metric tooltip positioning

* Add tests

* add cnapshots

* Fix up tests

* Recommit snapshots

* Fix snapshot test

* fix snapshot

---------

Co-authored-by: willhuang1997 <willhuang1997@gmail.com>

AnOctopus committed:
Enable fastReruns by default (#6200)

And adjust wording of config option to match.

kajarenc committed:
Cache storage abstraction (#6052)

Introduce the new CacheStorage and CacheStorageManager abstractions for @st.cache_data. This allows customizing the way that cache values are stored within Streamlit.

RuntimeConfig now takes a CacheStorageManager interface, and cache_data_api constructs the CacheStorage using that instance.

We ship with a concrete LocalDiskCacheStorage and LocalDiskCacheStorageManager
implementation, which just stores cache items in memory and on disk if persist=True is passed to st.cache_data (as before). This is what server.py uses when it initializes its Runtime instance.

We also implemented InMemoryCacheStorageWrapper class, which could serve as a facade and wrap any CacheStorage with an in_memory first layer cache.

Also, for testing purposes, DummyCacheStorage and DummyCacheStorageManagerimplemented.

Because the LocalDiskCacheStorageManager instance exists when a Runtime has been created, scripts that run in "raw" mode (i.e. streamlit._is_running_with_streamlit == False) no longer have access to that manager from Runtime. So in get_storage_manager in cache_data_api we fall back to DummyCacheStorageManager.

snehankekre committed:
List `str` as supported type for `show_spinner` in cache data docs (#6213)

Fixes #6207.

Co-authored-by: sfc-gh-jcarroll <joshua.carroll@snowflake.com>

lukasmasuch committed:
Pin altair version to <5 (#6219)

vdonato committed:
Fix retrieving messages from the server's ForwardMsgCache (#6210)

vdonato committed:
Make ping and websocket timeouts far more forgiving (#6212)

sfc-gh-mnowotka committed:
Merge pull request #6059 from streamlit/tasks/update-cloud-comm-whitelist-v2

Update ALLOWED_MESSAGE_ORIGINS to include address used for local development

sfc-gh-mnowotka committed:
Merge pull request #6042 from streamlit/sfc-gh-kbregula-add-line-numbers-to-st-code

Add line numbers to st.code

tconkling committed:
Remove unused `StyledForm` definition from Block.styled-components (#6240)

Trivial refactor: `Block/styled-components.ts` has an unused `StyledForm` definition. This PR just removes it!

vdonato committed:
Don't forget to stop script execution on websocket close (#6204)

There's a bug in the websocket reconnect logic that was implemented in #5856 where I forgot
to stop script execution in the websocket disconnect handler. This is fine with the vast majority
of scripts since most of them execute relatively quickly, but it does result in some weird behavior
with the (admittedly, rare and not-very-well-supported) use-case where a script is written to be
long-lived (via a while True loop or something similar). See #6166 for more information on the
bug.

This PR fixes this bug by simply requesting that the scriptrunner stop execution when on websocket
disconnect.

willhuang1997 committed:
[Snyk] Security upgrade vega from 5.22.1 to 5.23.0 (#6239)

* fix: frontend/package.json & frontend/yarn.lock to reduce vulnerabilities

The following vulnerabilities are fixed with an upgrade:
- https://snyk.io/vuln/SNYK-JS-VEGA-3338521

* Fix notices

* Rerun github actions

---------

Co-authored-by: snyk-bot <snyk-bot@snyk.io>
Co-authored-by: Maya Barnes <maya8617@gmail.com>

snehankekre committed:
`CachedStFunctionWarning` should include `experimental_allow_widgets`, not `suppress_st_warning` (#6217)

whitphx committed:
Use HTTPS to load external resources in the hello demo (#6232)

tconkling committed:
MetricsManager: de-singleton, interface-ize (#6241)

MetricsManager refactor, for upcoming lib/app split

- MetricsManager is no longer a singleton - there's no more `MetricsManager.current`. Instead, we pass our MetricsManager object to the few Streamlit components that consume it.
- MetricsManager is now an interface, and SegmentMetricsManager is our concrete implementation.

kcarnold committed:
Avoid showing emoji on non-UTF-8 terminals. (#6088)

* Avoid showing emoji on non-UTF-8 terminals.

It's unclear whether the remaining condition about Windows terminals is needed; it's not easy for me to test that. So I've left in that condition to be conservative.

Fixes: https://github.com/streamlit/streamlit/issues/2284

* Wrap long line as linter suggests.

* Add unit test

---------

Co-authored-by: Vincent Donato <vincent@streamlit.io>

sfc-gh-tszerszen committed:
Add support for tooltips on text elements (#6043)

sfc-gh-tszerszen committed:
Move typecheck and eslint to manual stage of pre-commit (#6171)

mayagbarnes committed:
Label Markdown Update  (#6163)

Update allowed/disallowed label markdown behavior (including colored text); address bugs/regressions; unwrap disallowed elements and render any text they contain.

snyk-bot committed:
fix: upgrade vega-lite from 5.6.0 to 5.6.1 (#6250)

Snyk has created this PR to upgrade vega-lite from 5.6.0 to 5.6.1.

See this package in npm:


See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/5fb5624f-b9e9-43fd-beb5-4c10d7c207df?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade @deck.gl/json from 8.8.23 to 8.8.24 (#6252)

Snyk has created this PR to upgrade @deck.gl/json from 8.8.23 to 8.8.24.

See this package in npm:


See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/5fb5624f-b9e9-43fd-beb5-4c10d7c207df?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade dompurify from 2.4.3 to 2.4.4 (#6253)

Snyk has created this PR to upgrade dompurify from 2.4.3 to 2.4.4.

See this package in npm:


See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/5fb5624f-b9e9-43fd-beb5-4c10d7c207df?utm_source=github&utm_medium=referral&page=upgrade-pr

vdonato committed:
Stop waiting for a browser to connect before opening a new tab (#6211)

snyk-bot committed:
fix: upgrade vega-embed from 6.21.0 to 6.21.2 (#6251)

Snyk has created this PR to upgrade vega-embed from 6.21.0 to 6.21.2.

See this package in npm:


See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/5fb5624f-b9e9-43fd-beb5-4c10d7c207df?utm_source=github&utm_medium=referral&page=upgrade-pr

Co-authored-by: willhuang1997 <willhuang1997@gmail.com>

sfc-gh-tszerszen committed:
Fix mypy cache data api and script runner (#6263)

tconkling committed:
Fix `make tstypecheck` (#6273)

`make tstypecheck` currently errors out with "No hook with id `typecheck` in stage `commit`". 

The fix is to pass `--hook-stage manual` to the precommit command. (The commit adds the hook-stake param to all make rules that use `pre-commit`, to prevent regressions in the future.)

sfc-gh-kbregula committed:
Enable no-else-return rule in ESLint (#6205)

sfc-gh-tszerszen committed:
Add step parameter to st.time_input (#6071)

tconkling committed:
`withMapboxToken`: add types (#6275)

`withMapboxToken` removes typing from its wrapped component. This PR adds proper typing - which is a readability nightmare, because HOCs are ugly.

With this change, TypeScript will raise a compile error if the wrapped component is instantiated with the wrong properties.

sfc-gh-kbregula committed:
Allow control of st.pyplot() width (#6067)

* Allow control of st.pyplot() width

tconkling committed:
`withFullscreenWrapper`: types (#6285)

Adds typing to our `withFullscreenWrapper` HOC

AnOctopus committed:
Support `st.exception` in testing framework (#6283)

tconkling committed:
SessionInfo: de-singleton-ify (#6294)

Remove the static `SessionInfo.current` and `last` singleton variables. `SessionInfo` now gets explicitly passed to components and managers that use it.

- `App` constructs the single `SessionInfo` instance that gets passed everywhere else.
- `SessionInfo.current`, `SessionInfo.last`, and `SessionInfo.isSet` are now member functions (instead of class functions)
- Lots of test code is modified: many of our tests would previously set the `SessionInfo.current` singleton during setup, which is no longer valid.
- To help with mocking SessionInfo instances for tests, there's a new `mocks.ts` utility file that has `mockSessionInfo` and `mockSessionInfoProps` helper functions.

whitphx committed:
Fix pre commit regex (#6281)

* Fix regex in .pre-commit-config.yaml

* Aggregate a common period from OR statements in regex

* Remove unused double hyphen from yarn command

* Remove unnecessary '$' symbol from the regex in .pre-commit-config.yaml

whitphx committed:
Refactoring App.test.tsx (#6270)

* Refactoring App.test.tsx

* Change an assertion on window.history.pushState

* Remove unnecessary mocks

whitphx committed:
Skip pushing a new history if the new page hash is the same as the current one (#6266)

* Skip pushing a new history if the new page hash is the same as the current one

* Use a destructured variable instead of object property access

* Rename currentPageScriptHash to prevPageScriptHash

tconkling committed:
withScreencast, withHostCommunication: types (#6286)

Types for `withScreencast` and `withHostCommunication`

sfc-gh-mnowotka committed:
Activate the no-unused-vars rule in eslint. (#6300)

* Initial cleanup of unused vars.

* More unused vars removed.

* Taking cate of $theme as well.

* Remove redundant variable.

snyk-bot committed:
fix: upgrade multiple dependencies with Snyk (#6262)

Snyk has created this PR to upgrade:
  - react from 16.13.1 to 16.14.0.
    See this package in npm: 
  - react-dom from 16.13.1 to 16.14.0.
    See this package in npm: 

See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/65ea6fb8-bb7d-46e3-979d-d2ee32dac40b?utm_source=github&utm_medium=referral&page=upgrade-pr

sfc-gh-kbregula committed:
Add st.divider (#6178)

* Add st.divider

* Update lib/streamlit/elements/markdown.py

Co-authored-by: Snehan Kekre <snehan.minerva@gmail.com>

---------

Co-authored-by: Snehan Kekre <snehan.minerva@gmail.com>

tconkling committed:
Pull ComponentRegistry's URL generation into an interface (#6307)

Move ComponentRegistry's URL-creation logic into a new "ComponentEndpointInfo" interface. This removes `ComponentRegistry`'s dependency on the Streamlit backend, and allows it to be used by other Streamlit frontends that may have different backend capabilities.

We ship `StreamlitComponentEndpoint` as the concrete implementation that talks to the vanilla Streamlit backend.

The PR also renames "ComponentRegistry.tsx" to "ComponentRegistry.ts" - this source file doesn't have any JSX syntax in it, so doesn't need the special extension.

kmcgrady committed:
Up version to 1.20.0 (#6284)

* Up version to 1.20.0

* Fix mypy cache data api and script runner (#6263)

---------

Co-authored-by: Ken McGrady <ken.mcgrady@snowflake.com>
Co-authored-by: Tomek Szerszeń <tomasz.szerszen@snowflake.com>

vdonato committed:
Replace @ts-ignore with @ts-expect-error and remove unused directives (#6308)

dependabot[bot] committed:
Bump webpack from 5.75.0 to 5.76.0 in /frontend (#6313)

Bumps [webpack](https://github.com/webpack/webpack) from 5.75.0 to 5.76.0.
- [Release notes](https://github.com/webpack/webpack/releases)
- [Commits](https://github.com/webpack/webpack/compare/v5.75.0...v5.76.0)

---
updated-dependencies:
- dependency-name: webpack
  dependency-type: direct:development
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump @sideway/formula from 3.0.0 to 3.0.1 in /frontend (#6309)

Bumps [@sideway/formula](https://github.com/sideway/formula) from 3.0.0 to 3.0.1.
- [Release notes](https://github.com/sideway/formula/releases)
- [Commits](https://github.com/sideway/formula/compare/v3.0.0...v3.0.1)

---
updated-dependencies:
- dependency-name: "@sideway/formula"
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

kajarenc committed:
ascii diagrams for CacheStorage* protocols, and for LocalDiskCacheStorage* concrete implementation (#6220)

Add ASCII diagrams for CacheStorage* protocols, and for LocalDiskCacheStorage* concrete implementation

mayagbarnes committed:
Add timeout to label-markdown spec (#6312)

The label-markdown e2e test is flaky because it isn't always able to load all the widgets within Cypress' default command timeout. This change increases the default timeout in the beforeAll hook to (hopefully) resolve.

mayagbarnes committed:
Add react-dropzone prop to address disabled file type behavior (#6315)

Updating to react-dropzone v12 caused st.file_uploader's File Selection Dialog to show all file types as available for selection, despite the extensions not being included in the type param. This overrides default use of File System Access API for react-dropzone. Additionally, .showOpenFilePicker() is still experimental and is not compatible with Firefox & Safari.

kajarenc committed:
Minor documentation fixes for cache storage (#6222)

Minor documentation fixes for cache storage

mayagbarnes committed:
Add nightly-preview deploy to nightly build (#6306)

Improve ability to test the nightly build as part of existing core-previews app - add step to nightly.yml to add whl url to nightly-preview branch on core-previews repo

vdonato committed:
Make using @ts-ignore illegal (#6314)

In #6308, we replaced usage of the `@ts-ignore` with the more strict but otherwise equivalent
`@ts-expect-error`. It seems unlikely we'll ever want to use the `@ts-ignore` directive in place of
`@ts-expect-error` since they're mostly equivalent except `@ts-expect-error` requires that the
line actually contain an error.

Ths PR enforces that this will be true going forward by adding a linter rule to make usage of `@ts-ignore`
illegal.

kajarenc committed:
Fix issue #6310 (#6321)

check that cache directory exists before trying to list files there.

we try to remove all files in the cache directory that start with the function key, whether clear is called for self.persist
storage or not, to avoid leaving orphaned files in the cache directory. This leads to a bug, when cache directory is not exist, but we try to list all files there (to filter only those that belong to the corresponding function).

The solution is to check that directory exists before trying to filter corresponding files there.

tconkling committed:
FileUploadClient: interface-ize endpoint (#6323)

Abstract the HTTP endpoint that `FileUploadClient` talks to, and encapsulate all Streamlit "endpoint logic" into a `StreamlitEndpoints` interface.

- Adds a new top-level interface, "StreamlitEndpoints", that handles both Custom Component and FileUploader endpoint logic.`ComponentRegistry` and `FileUploadClient` both take a `StreamlitEndpoints` instance.
- `DefaultStreamlitEndpoints` is our default concrete implementation of this new interface.
- A couple classes and interfaces are now gone, as a result:
-- `HttpClient` was the base class for `FileUploadClient`. It handled setting the CSRF token on file upload requests. This now lives in `DefaultStreamlitEndpoints.csrfRequest`
-- `StreamlitComponentEndpoint` is now folded into `DefaultStreamlitEndpoints`

AnOctopus committed:
Remove default values from internal class reprs

Using field defaults for dataclasses, and certain falsey field values for all classes, to determine what values are worth excluding.

mayagbarnes committed:
Cypress Upgrade to 9.7.0 (#6322)

Upgrading our Cypress version from v6.9.1 to v9.7.0. This significantly improves our Cypress test flakiness due to Cypress verification timing out.

whitphx committed:
Add an event handler for popstate event to reflect user's back button being clicked (#6271)

* Add an event handler for popstate event to reflect user's back button clicking

* Use testing-library.waitFor()

* Determine whether a new history should be pushed or not based on the page name and the current url path

* Replace @ts-ignore with @ts-expect-error

* Extract pageName from the URL path in consideration of basePath in handleNewSession

tconkling committed:
ForwardMessageCache: interface-ize endpoint (#6326)

Moves `ForwardMessageCache`'s "message cache endpoint" functionality into our `StreamlitEndpoints` interface (similar to ComponentRegistry and FileUploadClient).

- `StreamlitEndpoints.fetchCachedForwardMsg` is the new interface method
- `DefaultStreamlitEndpoints` talks to the Streamlit server `/_stcore/message` endpoint

vdonato committed:
Add sqlalchemy mypy types to test-requirements.txt (#6329)

sfc-gh-kbregula committed:
Add ability to turn off anchors (#6158)

* Add ability to turn off anchors

* fixup! Add ability to turn off anchors

* fixup! fixup! Add ability to turn off anchors

sfc-gh-kbregula committed:
Add support for period type in st.table and st.dataframe (#5429)

* Add support for period type in st.table and st.dataframe

* Apply code reviews comments

* Update tests

* Fix linter

* Fix tests

* Update tests

* Update tests

* fixup! Update tests

lukasmasuch committed:
Add support for number and boolean types in categorical columns (#6248)

* Add support for number & bool type

* Fix js test

vdonato committed:
Allow users to set a secrets.toml file in their home directory (#6230)

vdonato committed:
Tweak no-else-return config options (#6343)

tconkling committed:
sendMessageToHost: no longer a global function (#6345)

`sendMessageToHost` delivers "guest to host" messages to the Streamlit app's parent iframe (if one exists). It's currently implemented as a global function. This PR removes the global function; callers must now take the function as a property.

(`withHostCommunicationHOC` already passes the function to its wrapped component, so this isn't a huge change.)

snyk-bot committed:
fix: upgrade flatbuffers from 1.11.0 to 1.12.0 (#6259)

Snyk has created this PR to upgrade flatbuffers from 1.11.0 to 1.12.0.

See this package in npm:
https://www.npmjs.com/package/flatbuffers

See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/ccf3f55f-6b5f-4e23-b966-8c4970354edb?utm_source=github&utm_medium=referral&page=upgrade-pr

snyk-bot committed:
fix: upgrade command-line-args from 5.0.2 to 5.2.1 (#6258)

Snyk has created this PR to upgrade command-line-args from 5.0.2 to 5.2.1.

See this package in npm:
https://www.npmjs.com/package/command-line-args

See this project in Snyk:
https://app.snyk.io/org/willhuang1997/project/ccf3f55f-6b5f-4e23-b966-8c4970354edb?utm_source=github&utm_medium=referral&page=upgrade-pr

tconkling committed:
AppContext docstrings (#6353)

- Added docstrings for each AppContext property
- Removed unused `AppContext.layout` property
- Moved AppContext into `AppContext.tsx`, from `AppContext/index.tsx`

kajarenc committed:
Add `.webp` to list of safe static file extensions (#6331)

Add `.webp` to list of SAFE_APP_STATIC_FILE_EXTENSIONS

vdonato committed:
Return whether a secrets.toml file is successfully parsed (#6333)

sfc-gh-tszerszen committed:
Improve deploy button (#6223)

vdonato committed:
Have util.calc_md5 also take bytes (#6358)

* Have util.calc_md5 also take bytes

* Help mypy out a bit

PaleNeutron committed:
Fix regression with query_params  (#6348)

* fix bug with query_params, relate to #6347

* fix test

---------

Co-authored-by: John Lyu <lvjunhong@citics.com>

sfc-gh-tteixeira committed:
Improve st.help (and st.write's usage of st.help!) (#5857)

* Improve st.help and use it more in st.write.

* Add more tests

* Move imports into tests, so only the specific tests fail if the imports fail.

* If st.help(obj_instance), sometimes show value for Class.

* Support 3.7

* Improve behavior inside exec() blocks.
(i.e. it works but doesn't show variable name)

* Show warning when e2e/scripts/st_help.py is run within an exec() block
(this typically only happens in PR previews)

* Small refactors.

* Support for Python 3.7

* Add Github action to update snapshots (still in testing)

* Store Cypress videos

* Remove old screenshots

* Fix issues mayagbarnes brought up.

* Get test name from `github.events` in workflow-level statement.

* Fix e2e test.

* Readd import I removed by accident

* Fix tests.

* Add offset to screenshots.

* Add snapshots

* Fix Python 3.7 test

* Test shouldn't depend on internal module for no reason.

* Remove snapshots so we can rebuild them.

* Update screenshots

* Remove screenshots that broke when I fixed the header issue. Need
update.

* Fix magic e2e test.

* Add snapshots for e2e tests that broke when I fixed the header isssue.

* Add support for st.help() without args

* Tiny CSS change to show a little bit of overflowing docs.

* Add test.

* Enclose inline code in reStructuredText within two backticks

* Improve legibility and comments.

* Comments

* Prettify tests.

* Only return first line of multiline, and add more tests

* Fix st.help header overflow when narrow.

* Update snapshots

* Make st_help test work in 3.7

* Remove unecessary stuff.

* Fix pure Python run of st_help.py

* Differentiate st.help() from st.help(None)

* Fix typo in docstring

* Properly upgrade DocString.proto

* Move "reserved" line near the previous location.

* Fix 3.7 test

* Fix tests

* Fix tests

* Fix off-by-one bug on 3.7 due to :-1 rather than :None

* Ok, final fix for tests???

---------

Co-authored-by: snehankekre <snehan.minerva@gmail.com>

tconkling committed:
FullScreenWrapper: add a type declaration for react context usage (#6364)

- declare our `context` type, so that its usage is typechecked
- remove unused `isFullscreen` static variable
- add explicit public/private modifiers to properties

vdonato committed:
Set a default for RuntimeConfig.cache_storage_manager (#6361)

tconkling committed:
StreamlitEndpoints.buildMediaURL (#6366)

Adds `StreamlitEndpoints.buildMediaURL` and updates `Audio.tsx` to use it. (This is the first of several PRs that will update existing components to use this new API.)

- We currently have a top-level function, `buildMediaUri`, that takes a `BaseUriParts` instance and handles constructing Streamlit-relative URLs for media files. It has some assumptions about how Streamlit endpoints are named that we want to move away from for the StreamlitView project.
- This PR adds `StreamlitEndpoints.buildMediaURL` to the StreamlitEndpoints interface, and replicates the current implementation in `DefaultStreamlitEndpoints`.
- `Audio.tsx` is updated to use the interface-ized version of the function.

AnOctopus committed:
Allow session state to only allow serializable items (#6165)

lukasmasuch committed:
Allow pytest unit test to be discoverable via IDE v2 (#6374)

tconkling committed:
Remove UriUtil.buildMediaUri (#6379)

- `Video`, `ImageList`, `Favicon`, and `DownloadButton` elements now all use the interface-ized `StreamlitEndpoints.buildMediaURL` function.
- `UriUtil.buildMediaUri` is gone

(The PR also has some drive-by test naming changes.)

lukasmasuch committed:
Add support for Pandas 2.0 (#6378)

* Fix conftest

* Fix issues in tests

* Don't check for deprecated indices

* Pin pandas version to 2.x

* Explicitly set range index in data mocks

* Don't use float index

* Don't use old float index

* Delete old snapshot

* Remove deprecated index type

* Update snapshot

* Add test for old column types

* Remove version pinning of 2.x

* Add pin of major version

* Test CI with oldest pandas version

* Change pin

* Modify pin

* Pin version 1.0.x

* Change back to actual version

* Fix data editor test

schaumb committed:
Improve typing annotations for file_uploader and text_input (#6371)

* Add text_input 'type' parameter type hints

* Add file_uploader return value type hint

* Add missing import

* Enable any sequence on `file_uploader` type

* Replace NoneType to None

* Change `text_input` "type" parameter docstring

* Add `file_uploader` return cast

* Add type hint and cast to `_file_uploader`

* Update file_uploader.py

* Fix import

kajarenc committed:
supress warning when call `get_script_run_ctx` from `gather_metrics` decorator (#6384)

sfc-gh-tszerszen committed:
Rename query param ?testing to ?_stcore_testing (#6392)

tconkling committed:
StreamlitEndpoints.buildAppPageURL (#6386)

- Adds `StreamlitEndpoints.buildAppPageURL` to the StreamlitEndpoints interface.
- This function is used by `SidebarNav` to create URLs for app pages into multi-page apps
- `DefaultStreamlitEndpoints.buildAppPageURL` implements the app page URL logic that was previously in `SidebarNav`
- `AppContext.getBaseUriParts` is no longer used, and is now gone

vdonato committed:
Add AttrDict.to_dict() (#6387)

lukasmasuch committed:
Fix https test (#6399)

AnOctopus committed:
TextInput and TextArea testing classes (#6380)

* Don't run GC after every script run when testing

Running a gen 2 GC adds a lot of time relative to how little work the
script runs and tests are doing, and when testing we are not worried
about resource limits.

vdonato committed:
Relax session reconnect asserts (#6360)

snehankekre committed:
Embed example apps in `st.help` docstring (#6382)

* Embed example apps in `st.help` docstring

* Replace .DataFrame with .read_csv in st.help docstring example

kajarenc committed:
define SupportStr as a Protocol, instead of type alias to object (#6356)

define SupportStr as a Protocol, instead of type alias to object

lukasmasuch committed:
Support pandas styler for data editor (#6397)

sfc-gh-tteixeira committed:
Implement host->guest postMessage to update theme details (#6393)

* Implement host->guest postMessage that customizes the theme.

* Pipe new theme primitives into emotion theme.

* Update baseweb theme.

* Pipe font sizes into globalStyles. Replace absolute sizes with px values (since theme uses px) and keep relative sizes (rem) untouched.

* Adjust checkbox styles. Made checkbox squarer in the process.

* Fix numberinput styles. This component is built by us, not directly by baseweb.

* Propagate new styles to sidebar elements.

* Add ability to try out host->guest communication locally.

* Add tests.

* Fix tests.

* Fix widgets that got inadvertently changed before.

* No-op. Stop using px values in globalStyles when not necessary.

* Rename themeConfigObj -> themeInfo

* Improve hostframe.html instructions

* Fix return type for addPxUnit function

* Remove unused import

* Fix duplicated import

* Make react in scope and add return type

* Update ThemedApp and withHostCommunication tests

* Fix theme value propagation bug.

This addresses the radius propagation issue, and allows us to remove
hacky code that manually propagated the widgetBackgroundColor and
widgetBorderColor in ThemedSidebar.tsx.

* Wrap ThemedApp callbacks in useCallback().

* Update hostCommunication test

* Fix App.tsx test

* Update payload for themeInfo in hostCommunication

* Remove localhost allowedOrigin

* Resolve lint & update test instructions

---------

Co-authored-by: Juan Martin Garcia <juan.garcia@snowflake.com>
Co-authored-by: Maya Barnes <maya8617@gmail.com>

Rahs0601 committed:
Fix clipPath in svg test to let it render in Firefox  (#6375)

AnOctopus committed:
Move semver to dev dependency, temporarily pin (#6418)

* Temp pin pillow version

Until we have time to investigate the changes that caused tests to
break.

willhuang1997 committed:
ESLint: add "plugin:react-hooks/recommended", and fix issues  (#6401)

* Add fixes for eslint

* Add ignore

* Ignore for useTableSizer because it breaks tests

* fix ThemedApp and PlotlyChart eslint warnings (#6407)

And remove the William-specific `useTableSizer` TODO - that's a bigger tech debt that we can hand to someone else.

* Remove dependencies and add comment

* Remove extra package.json config

* Remove extra comma

* Revert package.json

---------

Co-authored-by: Tim Conkling <tconkling@gmail.com>

sfc-gh-kbregula committed:
Generate constraints files for all supported Python versions (#6234)

* Generate constraints files on CI

* Avoid the need to reconfigure protected branches

* fixup! Avoid the need to reconfigure protected branches

* Add documentation

* Extract methods

* Fix typo

AnOctopus committed:
Fix pillow rectangle generation in test (#6430)

This allows us to remove the upper bound on pillow, as I have determined
that the failure does not reflect a potential problem in our code, just
incidental to the test.

sfc-gh-kbregula committed:
Migrate to pip only and drop pipenv (#6265)

AnOctopus committed:
Printed / logged strings papercut fixes (#6391)

* Shrink widget id prefix

Make it less noisy in logs. The exact value doesn't really matter.

* Add clean repr to commonly printed dataclasses

sfc-gh-jrieke committed:
Typo (#6438)

Fixes typo in st.text_input's on_change param docstring.

kmcgrady committed:
Release 1.21.0 (#6444)

* Up version to 1.21.0

* Move semver to dev dependency, temporarily pin (#6418)

* Temp pin pillow version

Until we have time to investigate the changes that caused tests to
break.

* Fix pillow rectangle generation in test (#6430)

This allows us to remove the upper bound on pillow, as I have determined
that the failure does not reflect a potential problem in our code, just
incidental to the test.

---------

Co-authored-by: Ken McGrady <ken.mcgrady@snowflake.com>
Co-authored-by: Amanda Walker <amanda@amandawalker.io>

mayagbarnes committed:
Pass additional permission in nightly workflow (#6445)

Fix for nightly build - pass additional permission to python-versions workflow called in nightly workflow

AnOctopus committed:
Number input testing framework implementation (#6439)

AnOctopus committed:
Color picker test framework class (#6447)

sfc-gh-jgarcia committed:
Replace `:focus` with `:focus-visible` styles (#6373)

* Replace focus with focus-visible styles

* Update tests

kajarenc committed:
Change root logger to be in streamlit namespace (#6377)

* Change root logger to be in streamlit namespace
---------
Co-authored-by: David C <david@sstm2.com>

kajarenc committed:
Enrich communication between parent and child iframe (#6342)

When iframing Streamlit, this PR makes it possible for the host frame to send signals to Streamlit app to STOP_SCRIPT, RERUN_SCRIPT and to CLEAR_CACHE.

In opposite direction (from Streamlit iframe to parent iframe) send messages whenever the "script run state" changes.

Note that this is not meant to be a public API

This PR also adds functionality to frontend/hostframe.html to test host-guest communication described here. See that file for usage.

kajarenc committed:
Fix #6440: Add uuid to types we natively hash (#6459)

* Fix #6440 Add uuid to types we natively hash

UUID is a built-in Python type (lives in the standard library). This PR adds support for uuid hashing, so it would be possible to use uuid objects as parameters for functions decorated with @st.cache_data

tconkling committed:
Rename theme constants for better legibility (#6475)

The names "lightTheme" and "darkTheme" are overloaded in the codebase - "lightTheme" can refer to our top-level `ThemeConfig` light theme, or our Emotion-specific light theme, or our BaseUI-specific light theme, or the default BaseUI light theme.

With this PR, these are better differentiated:
- "lightTheme" is the top-level Streamlit `ThemeConfig` theme data, as before
- "emotionLightTheme" is the Emotion-specific theme name
- "baseuiLightTheme" is the BaseUI-specific theme name

willhuang1997 committed:
Empty mocks (#6476)

* Empty mocks

* Remove unnecessary import

* Fix lints

* Address comments

* Remove session info in constructor

* Implement changes

tconkling committed:
`Theme` -> `EmotionTheme` (#6477)

* `Theme` -> `EmotionTheme`

* update snap

tconkling committed:
Mock theme data in tests (#6479)

Some of our tests use concrete `lightTheme`/`darkTheme` theme implementations. When the upcoming lib/app frontend split happens, our component test files will no longer have access to these themes. 

This PR adds a new `mockTheme` theme, and changes our tests to use it instead.

sfc-gh-kbregula committed:
Use constraint files to install dependencies on PRs (#6396)

* Use constraint files to install dependencies on PRs

* fixup! Use constraint files to install dependencies on PRs

* fixup! fixup! Use constraint files to install dependencies on PRs

* fixup! fixup! fixup! Use constraint files to install dependencies on PRs

sfc-gh-kbregula committed:
Avoid storing GitHub token in .git directory (#6453)

sfc-gh-kbregula committed:
fixup! Avoid storing GitHub token in .git directory (#6453) (#6489)

sfc-gh-jgarcia committed:
Update multiselect icon (#6471)

* Update multiselect icon size and hover

* Remove old snapshots

* Add removed snapshots

* Remove other offending snapshots

* Add back offending snapshots

* Remove unused hover effect

* Update comment

sfc-gh-kbregula committed:
Restrict the setting of sensitive options by the CLI flag (#6376)

sfc-gh-kbregula committed:
Add config option to control the hamburger menu (#6174)

* Add config option to control the hamburger menu

* Update lib/streamlit/config.py

Co-authored-by: Snehan Kekre <snehan.minerva@gmail.com>

* fixup! Add config option to control the hamburger menu

---------

Co-authored-by: Snehan Kekre <snehan.minerva@gmail.com>

vdonato committed:
Update CODEOWNERS (#6496)

kmcgrady committed:
Update metadata description (#6503)

vdonato committed:
Autoformat yaml in .github dir and enforce formatting (#6497)

* Run prettier on the .github dir

* Enforce prettier is run on .github repo via precommit

* Use markdown list syntax instead of array literal

sfc-gh-kbregula committed:
Create tag with constraints files for new releases (#6495)

* Remove .github/actions/tag_and_repo action

* Create tag with constraints files for release

* fixup! Create tag with constraints files for release

* fixup! fixup! Create tag with constraints files for release

blackary committed:
No longer send unnecessary identify calls from frontend (#6464)

* No longer send unnecessary identify calls from frontend

* Fix length

* Fix length back, add back in standard vendor code

* Remove unused authorEmail

* Rename, add docstring

* Add test of save sending email

* Test with failing error

* Separate out passing and error tests and check logs

* Fix docstring

* Move tests, remove spaces

* Use dedicated library rather than requests

* Remove dup

* Back to manual method

* Add newline back

* Simplify, use json to stringify

* Switch from i to t

lukasmasuch committed:
Fix issue with deprecated asyncio test variable (#6517)

* Use new way to access async_io loop

* Add a change

* Change back

willhuang1997 committed:
Add Bokeh as a side effect import to modify globals instead of having it in index.html(#6512)

* Current

* Add new bokeh files but it's taking forever to yarn start

* Current

* Working with help of Jichan

* cleanup

* Readd tsconfig to fix linter

* Remove tsconfig because imports are followed

* Fix Bokeh Tests

* Stop checking for js files

* Revert it because it doesn't work

* Fix lint error

* Ignore vendor files for codeql

* cleanup bokeh test

* Fix naming

* Fix test by readding mockClear

* Fix test lint

* Fix codeql yaml lint

* Fix license and fix makefile

* Move bokeh min files to src/vendor and change imports

* Reclarify comments

* fix directory path for bokeh license in makefile

* Fix codeql

* Change comment to be more clear and concise

AnOctopus committed:
Date and Time test classes (#6501)

sfc-gh-kbregula committed:
Reorganize tests from streamlit_test.py (#6481)

sfc-gh-kbregula committed:
Dont use constraints file in scheduled builds (#6470)

vdonato committed:
Add tenacity as dependency (#6529)

* Add tenacity as dependency

* Also add upper bound

sfc-gh-kbregula committed:
Fix E2E image (#6524)

snehankekre committed:
Fix regression in visibility of `st.code`'s copy-to-clipboard button (#6498)

* Fix regression in `st.code` copy button and update `ElementNodeRenderer`

* Move copy button rendering and its container styles from ElementNodeRenderer.tsx and StreamlitMarkdown.tsx to StreamlitSyntaxHighlighter.tsx

* Add a conditional check to only render copy button when code block is not empty

* Remove the extra vertical space between code blocks caused by the added StyledCodeBlock

* fix: adjust spacing between code blocks in StyledCodeBlock

* Add E2E tests

* fixup! Add E2E tests

* Refactor `StyledCopyButton` styles to remove redundancies

---------

Co-authored-by: Kamil Bregula <kamil.bregula@snowflake.com>

connortann committed:
Add support for pandas 2.0 (#6507)

* Add support for pandas 2.0

* Update test constraint to use full version

* Pin pandas to <3.0

Co-authored-by: Lukas Masuch <Lukas.Masuch@gmail.com>

---------

Co-authored-by: Lukas Masuch <Lukas.Masuch@gmail.com>

lukasmasuch committed:
Update emojis to latest state (#6532)

blackary committed:
Clarify what telemetry data our backend stores (#6463)

* Send 0.0.0.0 for events from front-end

* Fix test

* Fix length

* Wrong branch

* Add comment explaining

* Use new object

lukasmasuch committed:
Fully remove email from new session message (#6516)

* Remove email property from NewSession proto

* Remove email from test mocks

* Add reserved

sfc-gh-jgarcia committed:
Fix code block font change (#6535)

lukasmasuch committed:
Add additional attributions (#6536)

* Add llm deps to attributions

* Move to top

* Add package

* Add transformers

* Add additional attributions

* Update list of attributions

* Update list of attributions

vdonato committed:
st.experimental_connection: The big merge (#6487)

`st.experimental_connection` is a new feature designed to help with the currently tedious process of configuring
and using connections to data sources inside of a Streamlit app. It provides a few things as part of its
public-facing API:

* The `st.experimental_connection` factory function, which helps developers with initializing a connection
   while following best practices that are often easy to overlook. Most notably, it ensures that connections
   created with it are cached so that they don't get reinitialized with every script run.
* Two first-party connections: `SQLConnection` and `SnowparkConnection`. These connections can easily
   be created from the `st.experimental_connection` factory function. They provide ways to work with the
   SQLAlchemy and Snowpark for Python libraries, respectively, with built-in support for Streamlit's secrets
   management mechanisms.
   * We'll be releasing additional connections that aren't shipped as part of the core library but are maintained
      by the Streamlit team. These connections may eventually be integrated into the core library after we're
      more confident in their quality and assuming there aren't dependency-related reasons we're unable to
      include them.
* For developers who want to create their own connections, we provide an `ExperimentalBaseConnection`
   abstract class to simplify the process of writing an `st.experimental_connection`-compatible connection
   implementation.

willhuang1997 committed:
Removing viz-1.8.0.min.js (#6520)

we remove Viz.js in the index.html as it's not really maintained. Before, we added it to allow Viz.js to run In a separate thread but many users don't use graphviz and the team recommends to remove it and then check the performance in a month for graphviz apps. When I tested it, it didn't seem too different of performance.

lukasmasuch committed:
Move pandas styler logic to dedicated module (#6543)

* Move pandas styler logic to dedicated module

* Fix typing

* Add license header

lukasmasuch committed:
Improve editing on touch devices for `st.data_editor` (#6548)

* Improve data_editor touch editing

* Add comment

snehankekre committed:
Replace `st.connection` with `st.experimental_connection` in docstring examples (#6553)

tconkling committed:
ESLint: use `--cache` flag (30x speedup!) (#6555)

`yarn lint`: use the ESLint `--cache` flag, so that only modified files get linted.

With a warm ESLint cache on my machine, this change reduces `yarn lint`'s running time from 26.98s -> 0.91s.

AnOctopus committed:
Clean up and reorganize element tree module (#6522)

Mostly alphabetizing classes, method overloads, and element creation dispatch

lukasmasuch committed:
Unify missing values to None in the returned datastructure by `st.data_editor`.  (#6544)

* Unify missing values for selected datastructures

* Fix comment

* Fix comment

lukasmasuch committed:
Add icon for editable columns in `st.data_editor` (#6550)

* Add icon for editable columns

* Add comments

* Add tests to check for existence of icon

* Add comment

* Remove e2e snapshots

* Add updated snapshots

lukasmasuch committed:
Show warning for unsafe integer cells in `st.dataframe` (#6549)

* Show error for unsafe integer cells

* Add a comment

* Add error value

* Extract as extra test

lukasmasuch committed:
Update dataframe column properties on frontend (#6554)

* Add new column props

* Fix use data loader test

* Remove help property

* Add column config changes in backend

* Remove help property

* Fix linter error

* Add test for required property

* Add check for error

* Add check for error cell

* Add test for default value

* Minor fix

* Fix arrow utils test

* Add required and default to test

* Fix unit test

* Apply review feedback

* Apply review comment

lukasmasuch committed:
Add support for cell and column header tooltips in the dataframe component (#6561)

* Add initial tooltip implementation

* Correct test name

* Expose tooltips hook

* Add tooltips to dataframe component

* Minor changes

* Add cleartooltip callback

* Try ot test the click to close behavior

* Remove click to close test

* Clear imports

* Add clear tooltip callback

* Add tests to tooltip hook

* Add license header

* Disable linter rule

* Use constant everywhere

* Add comment

lukasmasuch committed:
Add datetime column

sfc-gh-tszerszen committed:
Fix top padding on sidebar when embed is true (#6565)

tconkling committed:
Decouple MetricsManager from AppNode (#6557)

`AppNode` currently depends on `MetricsManager`. If we release the upcoming "StreamlitLib" project as is, StreamlitLib users will have to supply their own MetricsManager implementation. With this PR, `MetricsManager` no longer needs to be part of StreamlitLib.

- `AppNode` no longer knows about `MetricsManager`
- Instead, `App.handleDeltaMessage` sends Deltas to `MetricsManager`
- Removed several no-longer-necessary functions from the `MetricsManager` public API
- Added a bunch of previously-unimplemented `SegmentMetricsManager` tests

lukasmasuch committed:
Rename boolean and categorical columns (#6562)

* Rename boolean column to checkbox column

* Rename categorical to selectbox column

* Rename to selectbox

* Improve comment

* Additional renamings

* Fix selectbox test

* Add validate parameter to fix selectbox test

sfc-gh-tteixeira committed:
Make type=jpg and type=jpeg mean type=[jpg, jpeg] in file uploader. (#6559)

* Make type=jpg and type=jpeg mean type=[jpg, jpeg] in file uploader.

* Add support for more type pairs.

* Fix lower/upper case bug. Improve tests.

lukasmasuch committed:
Add a couple of updates

sfc-gh-tteixeira committed:
Clean up old experimental and beta commands (#6558)

* Remove `st.beta_*` since they're been deprecated since 2021.
* Remove `st.experimental_show` since it never worked all that well, and has almost no usage.

lukasmasuch committed:
Add datetime cell

lukasmasuch committed:
Merge remote-tracking branch 'remote/develop' into editableDatetime

lukasmasuch committed:
Add input validation capabilities to `st.data_editor` (#6563)

* Add validation option

* Implement validateCell

* Add validation support to text column

* Move validate cell to useDataEditor hook

* Remove unused imports

* Update comment

* Add more comments

* Add missing dependency

* Add tests for input validation

* Add tests for validate cell functionality

lukasmasuch committed:
Merge remote-tracking branch 'remote/develop' into editableDatetime

lukasmasuch committed:
Add datetime columns to index

lukasmasuch committed:
Add improved type parsing capabilities for `st.data_editor` (#6551)

* Add functionality to check underlying types

* Remove not-implemented types

* Add comment

* Some cleanup

* Add unit test

* Fix unit tests

* Finish unit test

* Add tests for index columns

* Remove type compatibility checks

* Remove refactoring

* Remove changes to column config object

* Remove final import

* Fix test issue

* Add dtype object to empty series for compatibility

* Add negative int and float to test

* Add a couple of comments about column data kind

lukasmasuch committed:
Merge remote-tracking branch 'remote/develop' into editableDatetime

lukasmasuch committed:
Use datetime columns based on arrow types

lukasmasuch committed:
Add a comment

lukasmasuch committed:
Add datetime types to backend

lukasmasuch committed:
Add custom cells to dataframe

lukasmasuch committed:
Update tests

lukasmasuch committed:
Fix utils unit test

lukasmasuch committed:
Try different name

lukasmasuch committed:
Add instruction to ignore eslint

lukasmasuch committed:
Add test cases

lukasmasuch committed:
Renaming

lukasmasuch committed:
Other fixes

lukasmasuch committed:
Convert step to number

lukasmasuch committed:
Add min / max tests

lukasmasuch committed:
Add additional comment

lukasmasuch committed:
Add tests for step size

lukasmasuch committed:
Add test for format parameter

lukasmasuch committed:
Support timezone via parameter

lukasmasuch committed:
Add test for timezone

lukasmasuch committed:
Add comments

lukasmasuch committed:
Add todo

lukasmasuch committed:
Change timestamp default parsing to seconds

lukasmasuch committed:
Adjust timestamp based on unit

lukasmasuch committed:
Add test for timestamp adjustment

lukasmasuch committed:
Add no null assertion

lukasmasuch committed:
Add a smarter way to detect timestamp unit

lukasmasuch committed:
Auto time value to date

tconkling committed:
Remove deprecated metrics (#6569)

- Removes metrics events that the data team is no longer tracking:
  - "deltaStats"
  - "themeStats"
  - "pageConfigChanged"
  - "createReport"
- Removes the `MetricsManager` interface. This was created for the StreamlitLib refactor, but it's no longer necessary - no "lib" code uses metrics anymore.

lukasmasuch committed:
Fix datetime parsing in arrow utils

AnOctopus committed:
Testing property syntax (#6566)

* Support `sidebar` and `main` aliases

* Add alias properties for all elements

lukasmasuch committed:
Minor improvements

lukasmasuch committed:
Fix issue with milliseconds

lukasmasuch committed:
Update comments

lukasmasuch committed:
Remove outdated snapshots

lukasmasuch committed:
Smart adaption of default format for time column based on step size

lukasmasuch committed:
Merge remote-tracking branch 'remote/develop' into editableDatetime

lukasmasuch committed:
Use ts-expect-error

lukasmasuch committed:
Add comment

lukasmasuch committed:
Add missing snapshots

lukasmasuch committed:
Regenerate snapshot

lukasmasuch committed:
Regenerate snapshots

lukasmasuch committed:
Remove outdated snapshot

lukasmasuch committed:
Update snapshots
